model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/UserCacheSession.java

```diff
@@ -681,13 +683,35 @@ public UserConsentModel getConsentByClient(RealmModel realm, String userId, Stri
         CachedUserConsents cached = cache.get(cacheKey, CachedUserConsents.class);
 
         if (cached == null) {
+            UserConsentModel consent = getDelegate().getConsentByClient(realm, userId, clientId);
+            List<CachedUserConsent> consents;
+
+            if (consent == null) {
+                consents = Collections.singletonList(new CachedUserConsent(clientId));
```

Comment:
Should this be a `Collections.emptyList` instead? (same below)
Empty list means that no consents exist for a particular client

Reply:
The cached entry is not based on the client but still on a global key. Here we make sure non-existent consents are cached on a per-client basis to avoid unnecessary queries to the db.

Reply:
Ah, now I see how it works. Thanks, I'm resolving this now.